pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -533,7 +535,7 @@ private boolean buildSegmentInternal(boolean forCommit) {
     RealtimeSegmentConverter converter =
         new RealtimeSegmentConverter(_realtimeSegment, tempSegmentFolder.getAbsolutePath(), _schema,
             _segmentZKMetadata.getTableName(), _segmentZKMetadata.getSegmentName(), _sortedColumn,
-            _invertedIndexColumns, _noDictionaryColumns);
+            _invertedIndexColumns, _noDictionaryColumns, _starTreeIndexSpec);
```

Comment:
this constructor has become quite big. We need to clean up this class - it can be part of another PR